### ISSUE TITLE
SC2199 warns on [[ ${ARRAY[@]:x:1} == ... ]]

### ISSUE THREAD

**Issue by floatingatoll** [koalaman/shellcheck#885]
#### For bugs
- Rule Id: SC2199
- My shellcheck version: online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```bash
#!/bin/bash
A=(1 2 3); if [[ ${A[@]:0:1} == '1' ]]; then true; fi
```

#### Here's what shellcheck currently says:

SC2199: Arrays implicitly concatenate in [[ ]]. Use a loop (or explicit * instead of @).

#### Here's what I wanted or expected to see:

No warning, since the array is sliced to a single element before implicit concatenation occurs.

**Comment by koalaman** [comment]
You're right in this particular case because the array is statically defined, but in general this is not the case: it will expand to 1 *or 0* elements.

Here's what happens in `[` which doesn't implicitly concatenate:

```
$ A=(); [ "${A[@]:0:1}" == "foo" ]  && echo ok
bash: [: ==: unary operator expected
```

If you want 0 elements to be considered an empty string, you should explicitly concatenate the 0 or 1 elements so that you always end up with a string to compare. Alternatively, you can use "${A[0]}" which is shorter and amounts to the same.

**Comment by floatingatoll** [comment]
ooo.

Thank you! Sorry to bother.

**Comment by floatingatoll** [comment]
(I suppose there's an opportunity for :x:1 warning specifically recommending the simpler syntax.)
